### PR TITLE
rocAL - PR 529 + updates

### PR DESCRIPTION
**Pull request by rrawther**
@kiritigowda : this has all the PR529 changes and few more fixes to set proper affinity for RPP nodes
@paveltc : can you please run unit-tests on this

### CODE REVIEW

File: amd_openvx_extensions/CMakeLists.txt [hunk 1]
```diff
@@ -63,7 +80,24 @@ if(GPU_SUPPORT AND OpenCL_FOUND AND BUILD_OPENCL)
         message("-- ${Red}WARNING:FFMPEG Not Found -- amd_media module excluded${ColourReset}")
     endif(FFMPEG_FOUND)
 else()
-    message("-- ${Red}WARNING:OpenCL Not Found -- amd_rpp, amd_loomsl, & amd_media modules excluded${ColourReset}")
+    message("-- ${Red}WARNING:GPU support/OpenCL/FFMPEGNot Found -- amd_loomsl, & amd_media modules excluded${ColourReset}")
+endif()
+
+if(GPU_SUPPORT AND "${BACKEND}" STREQUAL "HIP")
```
Comment (asalmanp) [inline amd_openvx_extensions/CMakeLists.txt:86]: the "${BACKEND}" STREQUAL "HIP" needs to be changed to HIP_FOUND as we already searched for HIP in line 41-56 above.

File: amd_openvx_extensions/CMakeLists.txt [hunk 2]
```diff
@@ -63,7 +80,24 @@ if(GPU_SUPPORT AND OpenCL_FOUND AND BUILD_OPENCL)
...
+        if(HIP_FOUND)
```
Comment (asalmanp) [inline amd_openvx_extensions/CMakeLists.txt:87]: no need to have HIP_FOUND here if we check it in the above line as I mentioned in my previous comment.

**Comment by rrawther** [comment]
@kiritigowda : Can we merge this PR? Pavel ran the unit-tests and it is OK to merge

File: amd_openvx_extensions/CMakeLists.txt (hunk 1 above)
Comment (rrawther) [reply amd_openvx_extensions/CMakeLists.txt]: Done

File: amd_openvx_extensions/CMakeLists.txt (hunk 2 above)
Comment (rrawther) [reply amd_openvx_extensions/CMakeLists.txt]: Done

**Comment by kiritigowda** [comment]
@shobana-mcw can you confirm that #529 is captured in this PR and close #529 ?

**Review by kiritigowda — APPROVED** [review]

**Review by asalmanp — APPROVED** [review]

**Comment by rrawther** [comment]
@kiritigowda : since we merged PR#539, please discard this PR

**Comment by kiritigowda** [comment]
@rrawther & @asalmanp - This PR is failing the [base build tests](https://github.com/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pull/539/checks?check_run_id=2917935893#step:5:87) - can you take a look at this?

```
 -- WARNING:GPU support/OpenCL/FFMPEGNot Found -- amd_loomsl, & amd_media modules excluded
88
  CMake Error at amd_openvx_extensions/CMakeLists.txt:98 (message):
89
    Could'nt find HIP.  Ensure that HIP is either installed in /opt/rocm/hip or
90
    the variable HIP_PATH is set.
91
  
92
  
93
  -- Configuring incomplete, errors occurred!
```

**Comment by paveltc** [comment]
@rrawther MIVisionX fails to build with OCL backend with this PR. Can you please take a look?

**Comment by shobana-mcw** [comment]
> @shobana-mcw can you confirm that #529 is captured in this PR and close #529 ?

Yes. PR 529 is merged in this PR. I will close PR 529.

**Comment by rrawther** [comment]
@kiritigowda: Just fixed the build error for OpenCL backend happened with the previous commit.
This should be ready to merge now

**Review by kiritigowda — APPROVED** [review]

**Pull request merged**